adapters/rest-adapter-core/src/main/java/uk/gov/justice/services/adapter/rest/envelope/RestEnvelopeBuilder.java

```diff
@@ -101,8 +102,17 @@ private JsonObject buildPayload() {
                 .map(JsonObjects::createObjectBuilder)
                 .orElse(Json.createObjectBuilder());
 
-        for (Map.Entry<String, String> entry : params.orElse(emptyMap()).entrySet()) {
-            payloadBuilder = payloadBuilder.add(entry.getKey(), entry.getValue());
+        for (Parameter param : params.orElse(emptyList())) {
```

Comment:
It's a builder class so if the withParams method is not called then the params variable is empty.
That's my intention to skip the loop if it's empty.